helpers/liquity.ts

```diff
@@ -63,3 +62,88 @@ export function liquityV2Exports(config: IJSON<LiquityV2Config>) {
   return { adapter: exportObject, version: 2 } as SimpleAdapter
 }
 
+const RedemptionEvent = 'event Redemption(uint _attemptedLUSDAmount, uint _actualLUSDAmount, uint _ETHSent, uint _ETHFee)'
+const BorrowingEvent = 'event LUSDBorrowingFeePaid(address indexed _borrower, uint _LUSDFee)'
+
+type LiquityV1Config = {
+  troveManager: string
+  holderRevenuePercentage?: number
+  protocolRevenuePercentage?: number
+  redemptionEvent?: string
+  borrowingEvent?: string
+}
+
+
+
+export const getLiquityV1LogAdapter: any = (config: LiquityV1Config): FetchV2 => {
+  const fetch: FetchV2 = async (fetchOptions) => {
+    const { createBalances, getLogs, api } = fetchOptions
+    const dailyFees = createBalances()
+
+    const redemptionEvent = config.redemptionEvent || RedemptionEvent
+    const borrowingEvent = config.borrowingEvent || BorrowingEvent
+
+    // Get brrower opertaor contract
+    const borrowerOperator = await api.call({ abi: 'address:borrowerOperationsAddress', target: config.troveManager })
+    
+    // redemtions fees
+    const redemptionLogs = await getLogs({
+      target: config.troveManager,
+      eventAbi: redemptionEvent,
+    })
+
+    // event LUSDBorrowingFeePaid(address indexed _borrower, uint _LUSDFee);
+    const borrowingLogs = await getLogs({
+      target: borrowerOperator,
+      eventAbi: borrowingEvent,
+    })
+
+    redemptionLogs.forEach((logs) => {
+      dailyFees.addGasToken(BigInt(logs[3]))
+    })
+
+    borrowingLogs.forEach((logs) => {
+      dailyFees.addUSDValue(BigInt(logs[1])/BigInt(1e18))
```

Comment:
there are two assumptions here, that the stablecoin is pegged to USD and has 18 decimals

safer to fetch the stablecoin and do `dailyFees.add(token, log.fee)`

Reply:
part or all of the borrow fees goes to stabilityPool? so that part should be treated as supply side revenue?

Reply:
I think it holder rev
Because:
- You earn ETH and LUSD just by holding and staking LQTY
- You’re not providing a service like LPs do (no active role, no risk of impermanent loss)
- Your rewards come from the protocol’s fee income, distributed to token holders (who stake)

Reply:
you are right